src/routes/courses.ts

```diff
@@ -10,12 +10,26 @@ interface Course {
   sections?: number[];
 }
 
-router.get('/courses', (req, res) => {
+interface Query {
+  subject?: string;
+  code?: string;
+  term?: string;
+}
+
+router.get('/courses/:term?/:subject?/:code?', (req, res) => {
+  const term = req.params.term;
+  const subject = req.params.subject;
+  const code = req.params.code;
+  let query: Query = {};
   let filter = {};
 
+  if (term) query.term = term;
+  if (subject) query.subject = subject;
+  if (code) query.code = code;
+
+  const search = req.query.search;
```

Comment:
I don't think this will work as expected. You should just be able to do:
```js
  const term = req.params.term;
  const subject = req.params.subject;
  const code = req.params.code;
  const query: Query = {};

  if (term) query.term = term;
  if (subject) query.subject = subject;
  if (code) query.code = code;

  // Execute the query and return a response
  Courses.find(query)
    .then((document: any) => {
      res.json(document);
    })
    .catch((error: any) => res.status(500).json(error));
```